tests/import_proxy_test.go

```diff
@@ -305,20 +314,110 @@ var _ = Describe("Import Proxy tests", func() {
+			Expect(err).To(BeNil())
+
+			url := fmt.Sprintf(utils.TinyCoreIsoRegistryURL, f.CdiInstallNs)
+			reg := cdiv1.DataVolumeSourceRegistry{
+				URL:           &url,
+				CertConfigMap: &cm,
+			}
+
+			By(fmt.Sprintf("Create new DataImportCron %s, url %s", cronName, *reg.URL))
+			dic := utils.NewDataImportCron(cronName, "5Gi", scheduleEveryMinute, dataSourceName, 1, reg)
+			dic.Annotations[controller.AnnPodRetainAfterCompletion] = "true"
+			retentionPolicy := cdiv1.DataImportCronRetainNone
+			dic.Spec.RetentionPolicy = &retentionPolicy
+
+			cron, err = f.CdiClient.CdiV1beta1().DataImportCrons(ns).Create(context.TODO(), dic, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Verify initial job succeeded")
+			initialJobName := controller.GetInitialJobName(cron)
+			Eventually(func() int32 {
+				job, err := f.K8sClient.BatchV1().Jobs(f.CdiInstallNs).Get(context.TODO(), initialJobName, metav1.GetOptions{})
+				if err != nil {
+					Expect(k8serrors.IsNotFound(err)).To(BeTrue())
+					return 0
+				}
+				return job.Status.Succeeded
+			}, timeout, pollingInterval).Should(Equal(int32(1)), "initial job is not succeeded")
+
+			By("Verify cronjob first job succeeded")
+			cronJobName := controller.GetCronJobName(cron)
+			Eventually(func() *metav1.Time {
+				cronjob, err := f.K8sClient.BatchV1().CronJobs(f.CdiInstallNs).Get(context.TODO(), cronJobName, metav1.GetOptions{})
+				if err != nil {
+					Expect(k8serrors.IsNotFound(err)).To(BeTrue())
+					return nil
+				}
+				if cronjob.Status.LastSuccessfulTime == nil {
+					return nil
+				}
+				return cronjob.Status.LastSuccessfulTime
+			}, timeout, pollingInterval).ShouldNot(BeNil())
+
+			By("Wait for DataImportCron UpToDate")
+			Eventually(func() bool {
+				var err error
+				cron, err = f.CdiClient.CdiV1beta1().DataImportCrons(ns).Get(context.TODO(), cronName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				condUpToDate := controller.FindDataImportCronConditionByType(cron, cdiv1.DataImportCronUpToDate)
+				return condUpToDate != nil && condUpToDate.Status == corev1.ConditionTrue
+			}, timeout, pollingInterval).Should(BeTrue(), "Timeout waiting for DataImportCron conditions")
+
+			By("Wait for CurrentImports update")
+			dvName := ""
+			Eventually(func() string {
+				cron, err = f.CdiClient.CdiV1beta1().DataImportCrons(ns).Get(context.TODO(), cronName, metav1.GetOptions{})
+				dvName = cron.Status.CurrentImports[0].DataVolumeName
+				Expect(dvName).ToNot(BeEmpty())
+				return dvName
+			}, timeout, pollingInterval).ShouldNot(BeEmpty())
+
+			By("Wait for datavolume succeeded")
+			err = utils.WaitForDataVolumePhase(f, ns, cdiv1.Succeeded, dvName)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking the importer pod information in the proxy log to verify if the requests were proxied")
+			verifyImporterPodInfoInProxyLogs(f, url, registryUserAgent, now, BeTrue)
```

Comment:
You can't use this function the way it is. This functions looks for the importer pod (which we know works already). We are interested in the poller working properly. So you have to either modify this function to pass in the prefix for the pod to look for, or use some other method of finding the poller pod.

Reply:
Fixing